File: src/opentimelineio/timedTextStyle.cpp
#include "opentimelineio/timedTextStyle.h"

namespace opentimelineio {
    namespace OPENTIMELINEIO_VERSION {

        TimedTextStyle::TimedTextStyle(const std::string &style_id,
                                       std::string text_alignment,
                                       std::string text_color,
                                       float text_size,
                                       bool text_bold,
                                       bool text_italics,
                                       bool text_underline,
                                       std::string font_family)
                : Parent(),
                  _style_id(style_id),
                  _text_alignment(text_alignment),
                  _text_color(text_color),
                  _text_size(text_size),
                  _text_bold(text_bold),
                  _text_italics(text_italics),
                  _text_underline(text_underline),
                  _font_family(font_family) {
        }

        TimedTextStyle::~TimedTextStyle() {}

        bool TimedTextStyle::read_from(Reader &reader) {
            return reader.read("style_id", &_style_id) &&
                   reader.read("text_alignment", &_text_alignment) &&
                   reader.read("text_color", &_text_color) &&
                   reader.read("text_size", &_text_size) &&
                   reader.read("text_bold", &_text_bold) &&
                   reader.read("text_italics", &_text_italics) &&
                   reader.read("text_underline", &_text_underline) &&
                   reader.read("font_family", &_font_family) &&
                   Parent::read_from(reader);
        }

        void TimedTextStyle::write_to(Writer &writer) const {
            Parent::write_to(writer);
            writer.write("style_id", _style_id);
            writer.write("text_alignment", _text_alignment);
            writer.write("text_color", _text_color);
            writer.write("text_size", _text_size);
            writer.write("text_bold", _text_bold);
            writer.write("text_italics", _text_italics);
            writer.write("text_underline", _text_underline);
            writer.write("font_family", _font_family);
        }

    }
}

File: src/py-opentimelineio/opentimelineio/schema/timed_text_style.py
from .. core._core_utils import add_method
from .. import _otio


@add_method(_otio.TimedTextStyle)
def __str__(self):
    return (
        'TimedTextStyle('
        '"{}", "{}", "{}", {}, {}, {}, {}, "{}")' .format(
            self.style_id,
            self.text_alignment,
            self.text_color,
            self.text_size,
            self.text_bold,
            self.text_italics,
            self.text_underline,
            self.font_family
        )
    )


@add_method(_otio.TimedTextStyle)
def __repr__(self):
    return (
        'TimedTextStyle('
        'style_id={}, '
        'text_alignment={}, '
        'text_color={}, '
        'text_size={}, '
        'text_bold={}, '
        'text_italics={}, '
        'text_underline={}, '
        'font_family={}'
        ')' .format(
            repr(self.style_id),
            repr(self.text_alignment),
            repr(self.text_color),
            repr(self.text_size),
            repr(self.text_bold),
            repr(self.text_italics),
            repr(self.text_underline),
            repr(self.font_family),
        )
    )